src/v/cluster/topics_frontend.cc

```diff
@@ -142,9 +142,17 @@ ss::future<std::vector<topic_result>> topics_frontend::create_topics(
          * (ntp_config::default_remote_delete) on the construction of
          * topic_properties(), so there is no need to overwrite it here.
          */
-        if (!tp.cfg.properties.shadow_indexing.has_value()) {
-            tp.cfg.properties.shadow_indexing
-              = _metadata_cache.get_default_shadow_indexing_mode();
+        auto& si = tp.cfg.properties.shadow_indexing;
+        if (!si.has_value()) {
+            si = _metadata_cache.get_default_shadow_indexing_mode();
+        }
+        if (si != model::shadow_indexing_mode::disabled) {
+            if (_features.local().should_sanction()) {
+                // TODO (ben): probably best to partition the requests and only
+                // fail the topics that have SI enabled
```

Comment:
would probably throw a log message in there as well about why

Reply:
+1 I think I would err on the side of being more verbose. Specifically, I think this would be a good place to call out why the feature is disabled (they don't have a valid enterprise license) and how they can get out of the error (get a valid license).